package/widevine/patches/0001-Disable-membarrier-for-Arris.patch

```diff
@@ -0,0 +1,21 @@
+diff --git a/oemcrypto/level3/clear_cache_function.h b/oemcrypto/level3/clear_cache_function.h
+index 2c6d1b9..7de5b5b 100644
+--- a/oemcrypto/level3/clear_cache_function.h
++++ b/oemcrypto/level3/clear_cache_function.h
+@@ -113,8 +113,14 @@ void clear_cache_function(void *page, size_t len) {
+ 
+ #if defined(__arm__) || defined(__aarch64__)
+ #  pragma message "(info): inserting membarrier_function calls."
+-  membarrier_function(MEMBARRIER_CMD_REGISTER_PRIVATE_EXPEDITED_SYNC_CORE);
+-  membarrier_function(MEMBARRIER_CMD_PRIVATE_EXPEDITED_SYNC_CORE);
++  // Note: On Arris, kernel 4.9 is being used. This widevine implementation 
```

Comment:
You're patching out memory barriers, which enforce an order of execution of all threads in this process. I think that they're here in order to enforce cache coherency after a flush operation. Regardless of the reason, I don't know the codebase too well but removing these barriers doesn't sound good, unless it's a patch provided by the WideVine team.

AFAIK for 4.9 kernel you can only use the global memory barrier, which in this case doesn't seem like the right choice. In my opinion you should either downgrade the version of widevine or bump the kernel version on Arris.

Reply:
@krp97 Thanks for reviewing the PR.
Downgrading version of widevine or bump the kernel version for Arris has external dependency. It would be great if Broadcom can provide higher kernel version.

Widevine itself here a check for kernel 4.16 and then only executes barrier here:
https://github.com/WebPlatformForEmbedded/widevine/blob/660ddf9422948c2851f8fa42d93e955317d99f90/oemcrypto/level3/clear_cache_function.h#L44

Reply:
@neeraj-deshpande why can't we downgrade the version of the widevine library? As far as I know, we used to use a different version present here: https://github.com/Metrological/widevine/commits/master and that worked fine for some time.

I realize that bumping the kernel version might be too big of an ask, but we should consider all options before merging this change.

Reply:
Adding Bram as he worked on Widevine 15.
@bramoosterhuis  is it possible to downgrade Widevine as suggested by @krp97  as Arris platform is using older kernel 4.9 whereas Widevine 15 expects kernel 4.16?